corehq/blobs/migrate.py

```diff
@@ -415,8 +415,12 @@ def get_doc_migrator(self, filename, date_range=None, **kw):
         migrator = super(BackendMigrator, self).get_doc_migrator(filename)
         return _migrator_with_worker_pool(migrator, self.reindexer, **kw)
 
-    def get_document_provider(self):
-        return SqlDocumentProvider(self.iteration_key, self.reindexer)
+    def get_document_provider(self, date_range=None):
+        iteration_key = self.iteration_key
+        if date_range:
+            (start, end) = date_range
+            iteration_key = '{}-{}-{}'.format(self.iteration_key, start, end)
+        return SqlDocumentProvider(iteration_key, self.reindexer)
```

Comment:
Just for sanity, @millerdev can you confirm `iteration_key` isn't used anywhere else other than here, where it need to be similar modified as above?

Reply:
The iteration key is used to allow an iteration over the full set of documents (or in this case, the subset of documents that fall within the date range) to be resumed. Multiple iterations can be performed concurrently (and their state will be maintained independently) as long as each one uses a different iteration key.

When running multiple migrations concurrently the person running the migration must be careful to use date ranges that do not overlap (this does not appear to be enforced automatically). Having two concurrent migrations running with overlapping date ranges could result in unexpected errors in the migration if both tried to migrate the same document at the same time. It's hard to evaluate quickly if that would cause data corruption—my initial guess is it should not, but I'm not very confident about that.